embedchain/config/apps/BaseAppConfig.py

```diff
@@ -1,14 +1,25 @@
 import logging
 
 from embedchain.config.BaseConfig import BaseConfig
+from embedchain.vectordb.vector_db import VectorDb
 
 
 class BaseAppConfig(BaseConfig):
     """
     Parent config to initialize an instance of `App`, `OpenSourceApp` or `CustomApp`.
     """
 
-    def __init__(self, log_level=None, embedding_fn=None, db=None, host=None, port=None, id=None):
+    def __init__(
+        self,
+        log_level=None,
+        embedding_fn=None,
+        db=None,
+        host=None,
+        port=None,
+        id=None,
+        db_type=None,
+        vector_dim: int = None,
```

Comment:
if its int, shouldn't its default value be zero?

Reply:
update: vector_dim: VectorDimensions = None,